### ISSUE TITLE
I have two questions:1) I train on my dataset for implement 2 classes when last layer sigmoid activate for binary cross entropy ,but the loss didn't convergence, 2) and the output shape is (N, C, 32, 32) when input shape is (N,C,128,128), can I unsample output to mask size（128,128） ？ 

### ISSUE THREAD

**Issue by liuchuanloong** [thomasjpfan/pytorch_refinenet#6]

**Comment by thomasjpfan** [comment]
1. There are a few reasons for losses not converging. Check if your gradients are misbehaving, if they are you can use gradient clipping to fix it.
2. [torch.nn.functional.interpolate](https://pytorch.org/docs/stable/nn.html?highlight=interpolate#torch.nn.functional.interpolate) scales tensors for you.

**Comment by liuchuanloong** [comment]
I try gradient clipping but did not work， how initial parameters except resnet layer ？ defaut pytorch？

**Comment by thomasjpfan** [comment]
Be default my refinenet implementation uses `torchvision.models.resnet101` for the resnet layer.

**Comment by thomasjpfan** [comment]
Make a plot of the gradients after each batch and see if they are exploding. Also plotting the loss after each batch would help in debugging your issue.

**Comment by yy9568** [comment]
@liuchuanloong Have you solved this problem?

**Comment by liuchuanloong** [comment]
@zhouyuangan  sorry, I just changed another scheme

**Comment by kingyj7** [comment]
FYI.
I also encountered the problem for loss not converging when I trained the model for 2 classes, and I deal with it just by decreasing the initial learning rate to 1e-4 or 5e-5.

**Comment by jingyzhang** [comment]
In my experiments, the learning rate 1e-5 seems too high to ensure stable loss convergence, where after several epochs the DiceLoss would increase to 1.00 and then does not decrease any more. Maybe a much smaller learning rate, e.g., 5e-6, can relieve this problem to some extent.

**Comment by kingyj7** [comment]
信件已收到:)
